### PR TITLE
python3Packages.pytenable: init at 1.2.8

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python library for the Tenable.io and TenableSC API

https://github.com/tenable/pyTenable

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117898` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python38Packages.pytenable</li>
    <li>python38Packages.python-box</li>
    <li>python38Packages.requests-pkcs12</li>
    <li>python38Packages.restfly</li>
    <li>python39Packages.pytenable</li>
    <li>python39Packages.python-box</li>
    <li>python39Packages.requests-pkcs12</li>
    <li>python39Packages.restfly</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117898` at 3cd07b42 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pytenable</li>
    <li>python38Packages.python-box</li>
    <li>python38Packages.requests-pkcs12</li>
    <li>python38Packages.restfly</li>
    <li>python39Packages.pytenable</li>
    <li>python39Packages.python-box</li>
    <li>python39Packages.requests-pkcs12</li>
    <li>python39Packages.restfly</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `arrow`.
Near pkgs/development/python-modules/requests-pkcs12/default.nix:2:3:
```
  |
2 | , arrow
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytenable/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  checkInputs = [
+    responses
+    pytest-datafiles
+    pytest-vcr
+    pytestCheckHook
+  ];
+
+  disabledTests = [
+    # Disable tests that requires a Docker container
```
Comment (dotlambda) [inline pkgs/development/python-modules/pytenable/default.nix:48]:
```suggestion
    # Disable test that requires a Docker container
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pytenable/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pytenable/default.nix:48]: `test_uploads_docker` matches the patterns for multiple tests.

Comment (fabaff) [reply pkgs/development/python-modules/pytenable/default.nix:48]: Tests are now all listed.

File: pkgs/development/python-modules/python-box/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    owner = "cdgriffith";
+    repo = "Box";
+    rev = version;
+    sha256 = "0fhmkjdcacpwyg7fajqfvnv3n9xd9rxjdpvi8z3j73a1gls36gf4";
+  };
+
+  propagatedBuildInputs = [
+    msgpack
+    pyyaml
+    ruamel_yaml
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-box/default.nix:27]:
```suggestion
    ruamel-yaml
```

that alias should be in master by now.

File: pkgs/development/python-modules/pytenable/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pytenable/default.nix:48]: Oh didn't know. I don't have a preference for listing all of them.
```suggestion
    # Disable tests that require a Docker container
```

File: pkgs/development/python-modules/python-box/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/python-box/default.nix:27]: Looks like that I have to wait till it shows up.

File: pkgs/development/python-modules/pytenable/default.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  buildInputs = [
+    defusedxml
+    marshmallow
+    python-box
+    python-dateutil
+    requests
+    requests-pkcs12
+    restfly
+    semver
```
Comment (jonringer) [inline pkgs/development/python-modules/pytenable/default.nix:37]: the python packages should go into propagatedBuildInputs

File: pkgs/development/python-modules/python-box/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/python-box/default.nix:27]: If it is not in master already we can ignore it for now.

File: pkgs/development/python-modules/pytenable/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/pytenable/default.nix]: Thanks, changed.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 117898 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>8 packages built:</summary>  <ul>    <li>python38Packages.pytenable</li>    <li>python38Packages.python-box</li>    <li>python38Packages.requests-pkcs12</li>    <li>python38Packages.restfly</li>    <li>python39Packages.pytenable</li>    <li>python39Packages.python-box</li>    <li>python39Packages.requests-pkcs12</li>    <li>python39Packages.restfly</li>  </ul></details>

**Pull request merged**